File: python/kindergarten-garden/kindergarten_garden.py
STUDENTS = (
    'Alice', 'Bob', 'Charlie', 'David',
    'Eve', 'Fred', 'Ginny', 'Harriet',
    'Ileana', 'Joseph', 'Kincaid', 'Larry',
)

PLANTS = {
    'C': 'Clover', 'G': 'Grass', 
    'R': 'Radishes', 'V': 'Violets',
}


class Garden:
    """Class to represent a kindergarten garden."""
    def __init__(self, diagram, students=STUDENTS):
        self.rows = diagram.split()
        kids = sorted(students)
        sets = zip(*([iter(self.rows[0])]*2
                   + [iter(self.rows[1])]*2))
        self.cups = {s:p for s,p in zip(kids, sets)}

    def plants(self, student):
        """Returns the plants a student is responsible for."""
        return [PLANTS[plant] for plant in self.cups[student]]

    def plants_all_students(self):
        """Returns all student plant responsibilities."""
        return {s:[PLANTS[p] for p in self.cups[s]]
                for s in self.cups}

    def plants_in_row(self, row=1):
        """Returns all the plants present in a row."""
        return [PLANTS[plant] for plant in self.rows[row-1]]

    def count_plant(self, plant='C'):
        """Returns how many times a plant has been planted."""
        return (PLANTS[plant],
                sum(row.count(plant) for row in self.rows))

File: csharp/kindergarten-garden/KindergartenGarden.cs
using System;
using System.Collections.Generic;
using System.Linq;

public enum Plant
{
    Violets,
    Radishes,
    Clover,
    Grass
}

public class KindergartenGarden
{
    private static readonly string[] Names =
    {
        "Alice", "Bob", "Charlie", "David", "Eve", "Fred",
        "Ginny", "Harriet", "Ileana", "Joseph", "Kincaid", "Larry",
    };

    private readonly Plant[] plantRowOne;
    private readonly Plant[] plantRowTwo;

    public KindergartenGarden(string diagram)
    {
        string[] diagramSplit = diagram.Split('\n');
        plantRowOne = ConvertRow(diagramSplit[0]);
        plantRowTwo = ConvertRow(diagramSplit[1]);
    }

    public IEnumerable<Plant> Plants(string student)
    {
        int leftPlantPosition = Array.IndexOf(Names, student) * 2;
        int rightPlantPosition = leftPlantPosition + 1;

        yield return plantRowOne[leftPlantPosition];
        yield return plantRowOne[rightPlantPosition];
        yield return plantRowTwo[leftPlantPosition];
        yield return plantRowTwo[rightPlantPosition];
    }

    private static Plant[] ConvertRow(string row)
    {
        return row.Select(ConvertLetter).ToArray();
    }

    private static Plant ConvertLetter(char letter)
    {
        return letter switch
        {
            'V' => Plant.Violets,
            'R' => Plant.Radishes,
            'C' => Plant.Clover,
            'G' => Plant.Grass,
            _ => throw new ArgumentException($"Invalid letter: {letter}."),
        };
    }
}
